test/MerkleLib.Proofs.ts

```diff
@@ -0,0 +1,109 @@
+import { expect } from "chai";
+import { merkleLibFixture } from "./MerkleLib.Fixture";
+import { Contract, BigNumber } from "ethers";
+import { MerkleTree } from "../utils/MerkleTree";
+import { ethers } from "hardhat";
+
+interface PoolRebalance {
+  leafId: BigNumber;
+  chainId: BigNumber;
+  tokenAddresses: string[];
+  bundleLpFees: BigNumber[];
+  netSendAmount: BigNumber[];
+  runningBalance: BigNumber[];
+}
+
+interface DestinationDistribution {
+  leafId: BigNumber;
+  chainId: BigNumber;
+  amountToReturn: BigNumber;
+  l2TokenAddress: string;
+  refundAddresses: string[];
+  refundAmounts: BigNumber[];
+}
+
+function randomBigNumber() {
+  return ethers.BigNumber.from(ethers.utils.randomBytes(31));
+}
+
+function randomAddress() {
+  return ethers.utils.hexlify(ethers.utils.randomBytes(20));
+}
```

Comment:
should we put these in the `utils` file? we've been doing something similar to this for other tests with common utils.

Reply:
We can!

Reply:
Done.